test/jdk/java/nio/Buffer/CharBufferAsCharSequenceTest.java

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.CharBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PrimitiveIterator.OfInt;
+import java.util.Random;
+
+import jdk.test.lib.RandomFactory;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/*
+ * @test
+ * @key randomness
+ * @library /test/lib
+ * @build jdk.test.lib.RandomFactory
+ * @summary tests the CharBuffer implementations behaving as CharSequence in various states (postion, limit, offset)
+ * @run junit CharBufferAsCharSequenceTest
+ */
+public class CharBufferAsCharSequenceTest {
+
+    private static final Random RAND = RandomFactory.getRandom();
+    private static final int SIZE = 128 + RAND.nextInt(1024);
+
+    private static char[] randomChars() {
+        char[] chars = new char[SIZE];
+        for (int i=0; i<SIZE; ++i) {
+            chars[i] = (char) RAND.nextInt();
+        }
+        return chars;
+    }
+
+    private static CharBuffer randomizeRange(CharBuffer cb) {
+        int mid = cb.capacity() >>> 1;
+        int start = RAND.nextInt(mid + 1); // from 0 to mid
+        int end = mid + RAND.nextInt(cb.capacity() - mid + 1); // from mid to capacity
```

Comment:
Same comment as above concerning the two-arg `nextInt`.

Reply:
done - also made sure there is a minimum "remaining" between position and limit
https://github.com/openjdk/jdk/pull/26282/commits/394832a472a03dc015ab857a679506b4e8ca34e6